sdks/java/harness/src/test/java/org/apache/beam/fn/harness/control/BeamFnControlClientTest.java

```diff
@@ -164,6 +175,88 @@ public void testDelegation() throws Exception {
     }
   }
 
+  @Test
+  public void testJavaErrorResponse() throws Exception {
+    BlockingQueue<StreamObserver<BeamFnApi.InstructionRequest>> outboundServerObservers =
+        new LinkedBlockingQueue<>();
+    BlockingQueue<Throwable> error = new LinkedBlockingQueue<>();
+    CallStreamObserver<BeamFnApi.InstructionResponse> inboundServerObserver =
+        TestStreams.<BeamFnApi.InstructionResponse>withOnNext(
+                response -> fail(String.format("Unexpected Response %s", response)))
+            .withOnError(error::add)
+            .build();
+
+    Endpoints.ApiServiceDescriptor apiServiceDescriptor =
+        Endpoints.ApiServiceDescriptor.newBuilder()
+            .setUrl(this.getClass().getName() + "-" + UUID.randomUUID().toString())
+            .build();
+    Server server =
+        InProcessServerBuilder.forName(apiServiceDescriptor.getUrl())
+            .addService(
+                new BeamFnControlGrpc.BeamFnControlImplBase() {
+                  @Override
+                  public StreamObserver<BeamFnApi.InstructionResponse> control(
+                      StreamObserver<BeamFnApi.InstructionRequest> outboundObserver) {
+                    Uninterruptibles.putUninterruptibly(outboundServerObservers, outboundObserver);
+                    return inboundServerObserver;
+                  }
+                })
+            .build();
+    server.start();
+    try {
+      ManagedChannel channel =
+          InProcessChannelBuilder.forName(apiServiceDescriptor.getUrl()).build();
+
+      EnumMap<
+              BeamFnApi.InstructionRequest.RequestCase,
+              ThrowingFunction<BeamFnApi.InstructionRequest, BeamFnApi.InstructionResponse.Builder>>
+          handlers = new EnumMap<>(BeamFnApi.InstructionRequest.RequestCase.class);
+      handlers.put(
+          BeamFnApi.InstructionRequest.RequestCase.REGISTER,
+          value -> {
+            throw new Error("Example Error");
```

Comment:
easier to debug if we know that this has the word `Test` within it so that we know its generated. `Example Error` -> `Test Error`,

Reply:
Done.